### PR TITLE
Social: Hide settings button on Simple sites on the auto-convert notice

### PR DESCRIPTION
**Pull request by gmjuhasz**
This makes it so on WPCOM sites the auto-conversion notice should not have the `Change settings` button, as they cannot turn the feature off as of now.

## Proposed changes:

* Added a prop to be able to hide the settings button 58609dd3721c8de62e593ac5698d7b89dca2a22d
* Added the check to hide the settings on Simple sites 0ac3ec30e937a1564e4f53db0b6c542965eaed89
* Added some tests for the component to increase our coverage 💪 5487c45bec5530f9bf81c483d9015f57d71e512b
* Since we hid the notice before I unhid it here 512744355d82b4c2d1adb1f348ce3756ab6b4d6b

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/orgs/Automattic/projects/733/views/2?pane=issue&itemId=46760258
## Does this pull request change what data or activity we track or use?

No
## Testing instructions:

* Until D131678-code is merged, you need to follow the Test plan on that patch and check out that patch on your sandbox.
  * On your `wpsh` sandbox check out the branch `arc patch D131678`
  * With Jetpack downloader download this PR onto the checked out branch with `bin/jetpack-downloader test jetpack add/social-conversion-notice-wpcom`.
  * When prompted, say not to for checking out trunk, then say yes in the next question to continue.
  
Now you should have both changes applied.

* Make sure you have an Instagram Business connection
* Start editing a new post and add an invalid but convertible media as a featured image - for example a 10Mb size jpg image
* You should see that the image will be converted, but the `Change settings` button should not appear.
* Clicking `Got it` and refreshing the page should not bring back the notice as it's dismissed.

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/form/index.js [hunk 1]
```diff
@@ -110,7 +111,9 @@ export default function PublicizeForm() {
 					<ShareCountInfo />
 					<BrokenConnectionsNotice />
 					<UnsupportedConnectionsNotice />
-					{ shouldAutoConvert && showValidationNotice && mediaId && <AutoConversionNotice /> }
+					{ shouldAutoConvert && showValidationNotice && mediaId && (
+						<AutoConversionNotice canChangeSettings={ ! isSimpleSite } />
```
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/components/form/index.js]: `isSimpleSite` is a function. May be you wanted to call it?
```suggestion
						<AutoConversionNotice canChangeSettings={ ! isSimpleSite() } />
```

File: projects/js-packages/publicize-components/src/components/form/test/auto-conversion-notice.test.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		return {
+			shouldShowNotice: jest.fn( () => shouldShowNoticeValue ),
+			dismissNotice,
+			NOTICES,
+		};
+	} );
+};
+
+describe( 'AutoConversionNotice', () => {
+	it( 'should render is shouldShowNotice is true', async () => {
```
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/components/form/test/auto-conversion-notice.test.js]:
```suggestion
	it( 'should render when shouldShowNotice is true', async () => {
```

File: projects/js-packages/publicize-components/src/components/form/test/auto-conversion-notice.test.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		mockUseDismissNotice( { shouldShowNoticeValue: true } );
+		const { container } = render( <AutoConversionNotice canChangeSettings={ true } /> );
+
+		expect( container ).not.toBeEmptyDOMElement();
+		await expect( screen.findByText( /Got it/i ) ).resolves.toBeInTheDocument();
+	} );
+
+	it( 'should not render is shouldShowNotice is false', async () => {
```
Comment (manzoorwanijk) [inline projects/js-packages/publicize-components/src/components/form/test/auto-conversion-notice.test.js]:
```suggestion
	it( 'should not render when shouldShowNotice is false', async () => {
```

File: projects/js-packages/publicize-components/src/components/form/index.js [hunk 4]
```diff
@@ -6,6 +6,7 @@
  * sharing message.
  */
 
+import { isSimpleSite } from '@automattic/jetpack-shared-extension-utils';
```
Comment (spsiddarthan) [inline projects/js-packages/publicize-components/src/components/form/index.js:9]: does `isSimpleSite` also cover Atomic sites?

Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/form/index.js:9]: That's a good question Sid, I think not, as there is an `isAtomicSite` function too, but it has to be tested!

File: projects/js-packages/publicize-components/src/components/form/index.js (hunk 1 above)
Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/form/index.js]: Fixen in f839ed86192252217553792de0398594c29319a9

File: projects/js-packages/publicize-components/src/components/form/test/auto-conversion-notice.test.js (hunk 2 above)
Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/form/test/auto-conversion-notice.test.js]: Fixed in f839ed86192252217553792de0398594c29319a9

File: projects/js-packages/publicize-components/src/components/form/test/auto-conversion-notice.test.js (hunk 3 above)
Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/form/test/auto-conversion-notice.test.js]: Fixed in f839ed86192252217553792de0398594c29319a9

File: projects/js-packages/publicize-components/src/components/form/index.js (hunk 4 above)
Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/form/index.js:9]: We tested this together on a pairing session

**Review by spsiddarthan — APPROVED** [review]

**Pull request closed without merging**